app/plugin_key_value_store.go

```diff
@@ -68,6 +68,23 @@ func (a *App) CompareAndSetPluginKey(pluginId string, key string, oldValue, newV
 	return updated, nil
 }
 
+func (a *App) CompareAndSetPluginKeyWithExpiry(pluginId string, key string, oldValue, newValue []byte, expireInSeconds int64) (bool, *model.AppError) {
+	updated, err := a.CompareAndSetPluginKey(pluginId, key, oldValue, newValue)
+	if err != nil {
+		mlog.Error("Failed to compare and set plugin key value", mlog.String("plugin_id", pluginId), mlog.String("key", key), mlog.Err(err))
+		return updated, err
+	}
+
+	if updated {
+		err = a.SetPluginKeyWithExpiry(pluginId, key, newValue, expireInSeconds)
```

Comment:
This does work, however it will cause a race condition where if there is an update that happens in between this call and the call above i.e. `CompareAndSetPluginKey`, then calling `SetPluginKeyWithExpiry` here will overwrite the update, which is not great. 

One idea that might work is to update `CompareAndSetPluginKey` to take a new parameter `expireInSeconds int64` and have `PluginStore.CompareAndSet` honour that value, similar to how `SetPluginKeyWithExpiry` is set up.

That way updating the plugin key value with expiry happens in one shot rather than two operations.

Reply:
Makes sense. Let me change it.

Reply:
Changed it to use existing CompareAndSet. Let me know what you think.